libtailscale/callbacks.go

```diff
@@ -23,13 +23,14 @@ var (
 	// onGoogleToken receives google ID tokens.
 	onGoogleToken = make(chan string)
 
-	// onDNSConfigChanged is notified when the network changes and the DNS config needs to be updated.
-	onDNSConfigChanged = make(chan struct{}, 1)
+	// onDNSConfigChanged is notified when the network changes and the DNS config needs to be updated. It receives the updated interface name.
+	onDNSConfigChanged = make(chan string, 1)
 )
 
-func OnDnsConfigChanged() {
+// ifname is the interface name retrieved from LinkProperties on network change, if any.
+func OnDnsConfigChanged(ifname string) {
```

Comment:
Unrelated to this PR, but the style is OnDNSConfigChanged per https://go.dev/wiki/CodeReviewComments#initialisms and matching the variable name `onDNSConfigChanged` in backend.go